crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2806,8 +2832,58 @@ impl Analyzer<'_, '_> {
                             }
                         }
                     }
+                    IntrinsicKind::Capitalize => {
+                        if let Some(value) = &value.raw {
+                            let ch = value.chars().next();
+
+                            if let Some(ch) = ch {
+                                if !ch.is_uppercase() {
+                                    return Err(ErrorKind::AssignFailed {
+                                        span: r.span(),
+                                        left: box Type::StringMapping(to.clone()),
+                                        right_ident: None,
+                                        right: box r.clone(),
+                                        cause: vec![],
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                    }
+                    IntrinsicKind::Uncapitalize => {
+                        if let Some(value) = &value.raw {
+                            let ch = value.chars().next();
+
+                            if let Some(ch) = ch {
+                                if !ch.is_lowercase() {
+                                    return Err(ErrorKind::AssignFailed {
+                                        span: r.span(),
+                                        left: box Type::StringMapping(to.clone()),
+                                        right_ident: None,
+                                        right: box r.clone(),
+                                        cause: vec![],
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                    }
                 }
-            },
+            }
+            Type::Union(union) => {
+                if union.types.iter().all(|v| v.is_str_like()) {
```

Comment:
Need to handle union better

Reply:
Nested unions are not allowed, so it may be enough